src/Core/Generator/SchemaGenerator.cs

```diff
@@ -0,0 +1,336 @@
+        /// </summary>
+        /// <param name="token">The JSON token to process.</param>
+        /// <param name="fieldName">The name of the field.</param>
+        /// <param name="parentType">The name of the parent type.</param>
+        /// <param name="isArray">Indicates whether the field is part of an array.</param>
+        /// <param name="parentArrayLength">The length of the parent array if applicable.</param>
+        /// <returns>The GraphQL type of the field.</returns>
+        /// <exception cref="InvalidOperationException">Thrown if the JSON token type is unsupported or if an array contains mixed types.</exception>
+
+        private string ProcessJsonToken(JsonElement token, string fieldName, string parentType, bool isArray, int parentArrayLength = 1)
+        {
+            bool isObjectType = false;
+            parentType = parentType.Pascalize();
+
+            string gqlFieldType = "String";
+            // If field name is "id" then it will be represented as ID in GQL
+            if (fieldName == "id")
+            {
+                gqlFieldType = "ID";
+            }
+            else
+            {
+                // Rest of the fields will be processed based on their type
+                switch (token.ValueKind)
+                {
+                    case JsonValueKind.Object:
+                    {
+                        string objectTypeName = fieldName.Pascalize();
+
+                        if (isArray)
+                        {
+                            objectTypeName = objectTypeName.Singularize();
+                        }
+
+                        // Recursively traverse nested objects.
+                        TraverseJsonObject(JsonDocument.Parse(token.GetRawText()), objectTypeName);
+
+                        isObjectType = true;
+                        gqlFieldType = objectTypeName;
+                        break;
+                    }
+                    case JsonValueKind.Array:
+                        gqlFieldType = ProcessJsonArray(token, fieldName, parentType.Singularize());
+                        break;
+
+                    case JsonValueKind.Number:
+                        if (token.TryGetInt32(out int _))
+                        {
+                            gqlFieldType = "Int";
+                        }
+                        else if (token.TryGetDouble(out double _))
+                        {
+                            gqlFieldType = "Float";
+                        }
+
+                        break;
+                    case JsonValueKind.True:
+                    case JsonValueKind.False:
+                        gqlFieldType = "Boolean";
+                        break;
+                    case JsonValueKind.Null or JsonValueKind.String:
+                        if (DateTime.TryParse(token.GetString(), DateTimeFormatInfo.InvariantInfo, DateTimeStyles.None, out DateTime _))
+                        {
```

Comment:
wouldn't this error out if this case is `JsonValueKind.Null` and then you try token.GetString() ?